pkg/platform/runtime/setup/setup.go

```diff
@@ -373,6 +414,22 @@ func (s *Setup) setupArtifact(buildEngine model.BuildEngine, a artifact.Artifact
 		return errs.Wrap(err, "Could not collect env info for artifact")
 	}
 
+	cnst := envdef.NewConstants(s.store.InstallPath())
+	envDef = envDef.ExpandVariables(cnst)
+	err = envDef.ApplyFileTransforms(filepath.Join(unpackedDir, envDef.InstallDir), cnst)
+	if err != nil {
+		return locale.WrapError(err, "runtime_alternative_file_transforms_err", "", "Could not apply necessary file transformations after unpacking")
+	}
+
+	// return s.moveToInstallPath(a, artifactName, unpackedDir, envDef, numFiles)
+	secondStageCh <- func() error { return s.moveToInstallPath(a, artifactName, unpackedDir, envDef, numFiles) }
```

Comment:
Maybe a slightly cleaner and more readable solution: https://github.com/faiface/mainthread/

Then we don't need to be passing any channel around either.

Reply:
I can do that...